File: Intermediate/9.Find the second largest number in an array/Solution.c
#include <stdio.h>
#include <limits.h>

int main() {
    int n, i, largest, secondLargest;
    
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    
    if (n < 2) {
        printf("Error: Array must have at least two elements.\n");
        return 1;
    }

    int arr[n];
    printf("Enter the elements of the array: ");
    for (i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    largest = secondLargest = INT_MIN;
    
    for (i = 0; i < n; i++) {
        if (arr[i] > largest) {
            secondLargest = largest;
            largest = arr[i];
        } else if (arr[i] > secondLargest && arr[i] != largest) {
            secondLargest = arr[i];
        }
    }

    if (secondLargest == INT_MIN) {
        printf("There is no second largest element.\n");
    } else {
        printf("The second largest number is: %d\n", secondLargest);
    }

    return 0;
}

File: Intermediate/9.Find the second largest number in an array/Solution.py
def find_second_largest(arr):
    if len(arr) < 2:
        print("Error: Array must have at least two elements.")
        return

    largest = second_largest = float('-inf')

    for num in arr:
        if num > largest:
            second_largest = largest
            largest = num
        elif num > second_largest and num != largest:
            second_largest = num

    if second_largest == float('-inf'):
        print("There is no second largest element.")
    else:
        print("The second largest number is:", second_largest)

if __name__ == "__main__":
    n = int(input("Enter the number of elements: "))
    if n < 2:
        print("Error: Array must have at least two elements.")
    else:
        arr = []
        print("Enter the elements of the array: ")
        for _ in range(n):
            arr.append(int(input()))
        find_second_largest(arr)
